src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs

```diff
@@ -157,6 +157,9 @@ public BitmapSource FinishImageProcessing(BitmapImage iconBitmapImage, Uri iconU
             // Only need to set this on one dimension, to preserve aspect ratio
             iconBitmapImage.DecodePixelWidth = DecodePixelWidth;
 
+            // Workaround for https://github.com/dotnet/wpf/issues/3503
+            iconBitmapImage.CreateOptions = BitmapCreateOptions.IgnoreColorProfile;
```

Comment:
This is interesting. If the image has a bad color profile or any other bad image data, then the catch at line 170 should have caught the exception. Instead, the whole devenv.exe process crashes.

Reply:
I think it is because the image needs to be downloaded from the URI which [delays the image creation](https://github.com/dotnet/wpf/blob/922458cd0094773d463a42afe94b869aa47bf82e/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/BitmapImage.cs#L100-L104).
In addition to that the [`BitmapImage` creates a `BitmapDecoder`](https://github.com/dotnet/wpf/blob/922458cd0094773d463a42afe94b869aa47bf82e/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/BitmapImage.cs#L358-L366) which uses a [`LateBoundBitmapDecoder`](https://github.com/dotnet/wpf/blob/922458cd0094773d463a42afe94b869aa47bf82e/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/BitmapDecoder.cs#L297) which uses the `BitmapDownload` helper class which uses a [separate thread for when the download finishes](https://github.com/dotnet/wpf/blob/ac9d1b7a6b0ee7c44fd2875a1174b820b3940619/src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/BitmapDownload.cs#L81). And that's where the exception is thrown and so it crashes the application.